src/core/math/matrixZipMap.ts

```diff
@@ -2,6 +2,30 @@ import * as R from 'ramda';
 
 type Matrix<T> = T[][];
 
+export const shallowClone: <T>(
+    m: Matrix<T>
+) => Matrix<T> = R.map(row => row.slice(0));
+
+export const traverse2 = <T1, T2, TR>(
+    a1: T1[],
+    a2: T2[],
+    fn: (d1: T1, d2: T2, i1: number, i2: number) => TR
+) => R.addIndex<T1>(R.forEach)((d1, i1) =>
+    R.addIndex<T2>(R.forEach)((d2, i2) =>
+        fn(d1, d2, i1, i2),
+        a2),
+    a1);
+
+export const traverseMap2 = <T1, T2, TR>(
+    a1: T1[],
+    a2: T2[],
+    fn: (d1: T1, d2: T2, i1: number, i2: number) => TR
+): TR[][] => R.addIndex<T1, TR[]>(R.map)((d1, i1) =>
+    R.addIndex<T2, TR>(R.map)((d2, i2) =>
+        fn(d1, d2, i1, i2),
+        a2),
+    a1);
+
```

Comment:
Utility. Apply `fn: (a,b) -> res` for all combinations of `(a,b)` of lists `A, B`.